tests/Http/ActionTest.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace Tests\Http;
+
+use Tests\TestCase;
+use Rogue\Models\Action;
+use Rogue\Models\Campaign;
+
+class ActionTest extends TestCase
+{
+    /**
+     * Test that a POST request to /actions creates a new action.
+     *
+     * POST /actions
+     * @return void
+     */
+    public function testCreatingAnAction()
+    {
+        // Create a campaign that the action will belong to.
+        $campaign = factory(Campaign::class)->create();
+        $actionName = $this->faker->sentence;
+
+        // Create an action;
+        $this->actingAsAdmin()->postJson('actions', [
+            'name' => $actionName,
+            'campaign_id' => $campaign->id,
+            'post_type' => 'photo',
+            'reportback' => 1,
+            'civic_action' => 0,
+            'scholarship_entry' => 1,
+            'noun' => 'things',
+            'verb' => 'done',
+        ]);
+
+        // Make sure the action is persisted.
+        $this->assertDatabaseHas('actions', [
+            'name' => $actionName,
+        ]);
+    }
+
+    /**
+     * Test that a GET request to /api/v3/actions returns an index of all campaigns.
```

Comment:
This might be too in the weeds but would this be better tested if there were actions assigned to multiple campaigns?

Reply:
also updated this!